### DISCUSSION TITLE
Wie funktioniert die Steuerung der WR durch OpenDTUonBattery ?

### DISCUSSION THREAD

**Discussion by expansiv** [hoylabs/OpenDTU-OnBattery#564] [Q&A]
Config:

- Shelly Pro 3EM
- OpenDTU-CC74A4 / v4.4.4 / 0.1.25 
- Zwei WR 3-phasig

die GUI der OpenDTU erlaubt drei Phasen einzeln abzufragen über Json:
/em:0/a_act_power   
/em:0/b_act_power   
/em:0/c_act_power   
Klickt man bei jeder Phase den Testbutton, wird mit "Sucess Power" der jeweils korrekte Wert von der Shelly für jede Phase angezeigt.
Unter "Live-Ansicht" Feld "Leistung/Netz" wird aber nur der Wert aus: /em:0/a_act_power  angezeigt = Phase 1.
Wahrscheinlich sind die Werte aus Phase2 und Phase3 nur für eine Batteriesteuerung vorgesehen und nicht für die Steuerung von WRs.?

Mit Json: /em:0/total_act_power  bekommt man die korrekte Summe über alle Shelly 3 Phasen egal, wie oft man den Befehl in den Felder der einzelnen Phasen einträgt. Habe ich das so richtig verstanden?

Eine weitere Frage bitte:
Nulleinspeisung:
Wenn der Shelly Wert aus /em:0/total_act_power 100 Watt wäre, würde die DTU dann beide WRs auf 50 Watt regeln?
Oder einen WR abschalten und den anderen auf 100 Watt regeln? Oder noch anders?

Noch eine Frage bitte:
Der Wert in der DTU unter Dynamic Power limiter Settings Angestrebter Netzbezug: -30
Ist das ein Offset das die Schwankungen zwischen den Nachregelungen abfängt und somit die Einspeisung verhindert?

Vielen Dank

**Comment by schlimmchen** [comment]
Bei dem Shelly kann ich nicht helfen. Ich versteh aber auch die Frage nicht so recht. Du hast scheinbar Probleme den korrekten Wert als Power Meter zu definieren?

> würde die DTU dann beide WRs auf 50 Watt regeln?

Der DPL in OpenDTU-OnBattery regelt (zur Zeit) nur genau einen Wechselrichter.

> Ist das ein Offset das die Schwankungen zwischen den Nachregelungen abfängt und somit die Einspeisung verhindert?

Das ist einfach nur der Wert, den der DPL versucht einzustellen als Netzbezug. Der hat keine tiefere Bedeutung außer jener, für die du diesen Wert benutzt. Also könnte die Antwort auf deine Frage "Ja" sein.

**Reply by expansiv** [reply]
Vielen Dank für die Antwort.

Ich habe mich zu umständlich ausgedrückt, bitte entschuldige. Deine Antwort dass, nur ein WR gesteuert wird hat es klar gestellt.

Mehrere WR's steuern: ist die Routine konzeptioniert? wie hoch schätzt Du den Aufwand schreiben, testen, implementieren?

Vielen Dank.

**Reply by schlimmchen** [reply]
Vor einigen Monaten habe ich mir darüber länger Gedanken gemacht und daher habe ich Ideen. Aufgeschrieben habe ich die nicht, aber wir hatten woanders darüber schon diskutiert. Erforderlich wäre es, dass man dem DPL mitteilt (per Einstellung in der WebApp), welche WR er regeln soll und woher die ihre Energie jeweils beziehen (Solarmodule vs. Batterie). Dann könnte man eine Logik implementieren, die verschiedene WR einstellt. Kompliziert wird es dann wenn WR mit Solarmodulen nicht genau das liefern, was der Algorithmus sich wünscht bzw. muss man dann herausfinden, was diese WR gerade liefern können, wenn man sie vorher begrenzt hat. Das wird dann lästig...

**Reply by expansiv** [reply]
Es ist ja schon eine Logik implementiert, die den WR ID0 auf- und abregelt. Ist das OO programmmiert? Bevor ich mich durch xxxxx Zeilen Code wühle, würde ich gerne den Aufwand abschätzen, Zeit ist ein Problem.

**Reply by Teufel24111991** [reply]
Ich würde das Thema gerne nochmals Hoch holen. 
Ich habe zwar keine 3Phasigen WR aber ich würde die Leistung zur Schonung der Komponenten einfach gleichmäßig auf alle Wechselrichter aufteilen. Sprich der Shelly misst 100W am Messpunkt und die Open DTU soll alle DPL Wechselrichter mit der Gleichen Leistung beschreiben. Also bei 2 WR mit je 50W. Bei 3 WR mit 33 Watt. 

Das müsste aus meiner Sicht doch einfach umsetzbar sein. So würden WR nachgelagerte Akkus gleichmäßig entladen. 
Ich könnte  mir das Gut mit einem Button alla Limit Gleichmäßig verteilen vorstellen.

**Reply by spcqike** [reply]
ganz so einfach ist das unter Umständen nicht, immerhin gibt es noch solar betriebene Wechselrichter und Wechselrichter an smarten Akkus, die nicht immer das geforderte Limit bereitstellen (bei 33W wohl eher unwahrscheinlich, aber stell dir mal 3x800W vor)

dazu kommt, dass die Effizienz der WR bei kleinen Lasten schlechter ist, als bei größeren. und zu kleine Lasten führen auch zu Problemen in der Regelung, dass der WR sein gesetztes Limit gar nicht korrekt trifft und teilweise nicht mehr richtig auf neue Befehle reagiert.

effizienter wäre es wohl, die Batterien einzeln zu entladen, und immer nur einen Wechselrichter laufen zu haben.

anyway, das ist hier scheinbar OT, gern kannst du den Vorschlag in #1409 einbringen.
